akka-sample-sharding-scala/src/main/scala/sample/sharding/TemperatureService.scala

```diff
@@ -0,0 +1,65 @@
+package sample.sharding
+
+import scala.concurrent.duration._
+import scala.util.Random
+
+import akka.actor.typed.{ ActorRef, Behavior }
+import akka.actor.typed.scaladsl.Behaviors
+import akka.cluster.sharding.typed.{ ClusterShardingSettings, ShardingEnvelope }
+import akka.cluster.sharding.typed.scaladsl.{ ClusterSharding, Entity, EntityTypeKey }
+import akka.cluster.typed.{ Cluster, Join }
+
+object TemperatureService {
+
+  // Update a random device
+  case object UpdateDevice extends Message
+  case object ReadTemperatures extends Message
+
+  val TypeKey: EntityTypeKey[Message] =
+    EntityTypeKey[Message]("Device")
+
+  def apply(): Behavior[Message] =
+    Behaviors.setup { context =>
+      val cluster = Cluster(context.system)
+      cluster.manager ! Join(cluster.selfMember.address)
+
+      val numberOfDevices = ClusterShardingSettings(context.system).numberOfShards
+      val random = new Random()
+
+      Behaviors.withTimers { timer =>
+        val deviceRegion: ActorRef[ShardingEnvelope[Message]] =
+          ClusterSharding(context.system).init(Entity(TypeKey)(createBehavior = entityContext =>
+            Device(entityContext.entityId)))
```

Comment:
My preference is to do that in the Device itself, with an `init` method that is called from here. Something like https://github.com/akka/akka-samples/pull/135/files#diff-a88d82b5450ab522ffbbf5480483de6bR84

Reply:
Should it be more like
```scala
def init(system: ActorSystem[_]): Unit = {
    ClusterSharding(system).init(Entity(TypeKey) { entityContext =>
      val numberOfShards = ClusterShardingSettings(system).numberOfShards
      val id = (math.abs(entityContext.entityId.hashCode) % numberOfShards).toString
      Device(id)
    })
  }
```
or
```scala
def init(system: ActorSystem[_]): Unit = {
    ClusterSharding(system).init(Entity(TypeKey)(entityContext =>
      Device(entityContext.entityId)))
  }
```

Reply:
Also the previous sample has the region in this Devices/TemperatureService not the Device. Why is doing init now in the entity better?

Reply:
grabbing the entityId from entityContext is the right way